### PR TITLE
Documentation - Update Common DG Sections

### PR DESCRIPTION
**Pull request by sharpstorm**
## Changes
This PR updates the **common** sections in the developer guide. It does not modify anything in the implementation and use cases section, which should be left for the individual.

A live preview is available [here](https://sharpstorm.github.io/tp/).

## Related Issues
None, this is a quality improvement PR.

### CODE REVIEW

**Review by dandaandaaaaaan — COMMENTED** [review]
generally lgtm, will let the others who are more familiar with the ui interactions to approve though

File: docs/diagrams/ModelAddressBookClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Person *--> Address
+Person *--> "*" Tag
+Tag *--> "1" "  Name  "
+
+Name -[hidden]right-> Phone
+Phone -[hidden]right-> Address
+Address -[hidden]right-> Email
+
+ModelManager -->"~* filtered" Person
+Appointment -> Person
```
Comment (KwanHW) [inline docs/diagrams/ModelAddressBookClassDiagram.puml:44]: Are we missing
`ModelManager --> "~* filtered" Tag` as `ModelManager` also keeps a filtered list of tags?

**Review by KwanHW — COMMENTED** [review]
LGTM. Just need to clarify 1 minor nit in the class diagram

File: docs/diagrams/ModelAddressBookClassDiagram.puml (hunk 1 above)
Comment (sharpstorm) [reply docs/diagrams/ModelAddressBookClassDiagram.puml]: You are right, fixed in the latest commit 0c95718.

**Review by Wei-Xinn — APPROVED** [review]
lgtm

**Review by KwanHW — APPROVED** [review]
LGTM

**Pull request merged**